examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+   "source": [
+    "class PrimeJavaScriptWidget(ipyreact.ReactWidget):\n",
+    "    prime_message = Any(\"Click the Button\").tag(sync=True)    # <- TODO: this message does not show up because prime_message is overwritten\n",
+    "    count = Int(0).tag(sync=True)\n",
+    "\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    function isPrimeNumber(n) {\n",
+    "    for (let i = 2; i < n; i++) {\n",
+    "        if (n % i === 0) {\n",
+    "        return \"No 🌐🧊🧊🧊\";\n",
+    "        }\n",
+    "    }\n",
+    "    return \"Yes 🌐✅✅✅\";\n",
+    "    }\n",
+    "\n",
+    "    export const MyUpdater = ({ count, prime_message}) => {\n",
+    "    prime_message = isPrimeNumber(count);\n",
+    "    return <span> {prime_message} </span>;\n",
+    "    };\n",
+    "\n",
+    "    export default function ({ on_count, count, prime_message}) {\n",
+    "    return (\n",
+    "        <div>\n",
+    "        <button onClick={() => on_count(count + 1)}>{count} times clicked</button>\n",
+    "        <br />\n",
+    "        <MyUpdater count={count} prime_message = {prime_message}/>\n",
+    "        </div>\n",
+    "    );\n",
+    "    }\n",
+    "    \"\"\"\n",
+    "primejs = PrimeJavaScriptWidget()\n",
+    "primejs"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "a038f153",
+   "metadata": {},
+   "source": [
+    "**SOON: call python functions from JavaScript**  \n",
+    "when https://github.com/widgetti/ipyreact/pull/8 is merged, it will also be possible to call python functions from JavaScript like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "abfa9586",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import ipyreact\n",
+    "# from traitlets import Int, Unicode\n",
+    "\n",
+    "# class Widget1(ipyreact.ReactWidget):\n",
+    "#     my_count = Int(0).tag(sync=True)\n",
+    "#     label = Unicode(\"Click me\").tag(sync=True)\n",
+    "\n",
+    "#     def on_my_python_function(self):\n",
+    "#         self.my_count = self.my_count + 1\n",
+    "#         self.label = f\"Clicked {self.my_count}\"\n",
+    "\n",
+    "#     _esm = \"\"\"\n",
+    "#         import * as React from \"react\";\n",
+    "\n",
+    "#         export default function({on_my_python_function, label}) {\n",
+    "#             return(\n",
+    "#             <div>\n",
+    "#                 <button onClick={() => on_my_python_function()}>\n",
+    "#                     {label}\n",
+    "#                 </button>\n",
+    "#             </div>\n",
+    "#             )\n",
+    "#         };\n",
+    "#     \"\"\"\n",
+    "            \n",
+    "# w1 = Widget1()\n",
+    "# w1"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "7476173b",
+   "metadata": {},
+   "source": [
+    "now that the examples are getting longer, it is a good idea to use separate files that contain our components."
```

Comment:
@paddymul :
I will add this sentence here at the end:
"Normal development practice is to have each resource (css, js, python) in a separate file. There are advantages to writing inline in Jupyter notebook cells. It's much quicker to get started, and examples are more compact with less file switching to follow along. Because of the self contained literate programming aspect, this tutorial will use the inline method.